src/Server.cpp

```diff
@@ -656,6 +626,12 @@ void cServer::KickUser(int a_ClientID, const AString & a_Reason)
 
 void cServer::AuthenticateUser(int a_ClientID, const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
+	if (GetNumPlayers() >= GetMaxPlayers())
```

Comment:
Another option would be to increment `m_PlayerCount` first and then check if they're allowed in.  That way accessing `m_PlayerCount` is still wait free.